src/js/stores/ChronosStore.js

```diff
@@ -0,0 +1,117 @@
+import {EventEmitter} from 'events';
+
+import AppDispatcher from '../events/AppDispatcher';
+import ChronosActions  from '../events/ChronosActions';
+import {
+  CHRONOS_JOBS_CHANGE,
+  CHRONOS_JOBS_ERROR,
+  VISIBILITY_CHANGE
+} from '../constants/EventTypes';
+import Config from '../config/Config';
+import JobTree from '../structs/JobTree';
+import {
+  REQUEST_CHRONOS_JOBS_ERROR,
+  REQUEST_CHRONOS_JOBS_ONGOING,
+  REQUEST_CHRONOS_JOBS_SUCCESS,
+  SERVER_ACTION
+} from '../constants/ActionTypes';
+
+import VisibilityStore from './VisibilityStore';
+
+let requestInterval;
+
+function startPolling() {
+  if (!requestInterval) {
+    ChronosActions.fetchJobs();
+    requestInterval = global.setInterval(
+      ChronosActions.fetchJobs,
+      Config.getRefreshRate()
+    );
+  }
+}
+
+function stopPolling() {
+  if (requestInterval) {
+    global.clearInterval(requestInterval);
+    requestInterval = null;
+  }
+}
+
+function handleInactiveChange() {
+  let isInactive = VisibilityStore.get('isInactive');
+  if (isInactive) {
+    stopPolling();
+  }
+
+  if (!isInactive && ChronosStore.shouldPoll()) {
+    startPolling();
+  }
+}
+
+VisibilityStore.addChangeListener(VISIBILITY_CHANGE, handleInactiveChange);
+
+class ChronosStore extends EventEmitter {
+  constructor() {
+    super(...arguments);
+
+    this.data = {
+      jobTree: {id: '/', items: []}
+    };
+
+    // Handle app actions
+    this.dispatcherIndex = AppDispatcher.register(({source, action}) => {
```

Comment:
I'm just chiming in to say that as @orlandohohmeier wrote, we seem to have a different strategy regarding components, @rcorral. It seems like all new React components are supposed to be ES6 classes (which is a PR-blocking requirement) even without a full-on refactoring process being required for preexisting non-ES6 components. 

Without having clear "consistency" guidelines written in a document these comments start feeling very arbitrary and ultimately become the cause of confusion, frustration and worst of all they waste our time. Let's please fix this once and for all.